cxbox-core/src/main/java/org/cxbox/core/dao/impl/MetadataUtils.java

```diff
@@ -207,37 +212,48 @@ public static Path getFieldPath(String fieldName, Root<?> root) {
 		return root.get(fieldName);
 	}
 
-	public static Predicate createPredicate(Root<?> root, ClassifyDataParameter criteria, CriteriaBuilder cb) {
+	public static Predicate createPredicate(Root<?> root, ClassifyDataParameter criteria, CriteriaBuilder cb,
+			String dialect) {
 		try {
 			Object value = criteria.getValue();
 
 			Path field = getFieldPath(criteria.getField(), root);
 
+			Expression filterPredicateForTime = TimeValueProvider.getFilterPredicate(cb, criteria, field, dialect, value);
```

Comment:
no! how will this work for new custom provider written on project!?! take provider from field. add method returning null to base class. call this method for ALL providers